routing/routing_tests/turns_generator_test.cpp

```diff
@@ -328,17 +328,17 @@ UNIT_TEST(TestRightmostDirection)
   TEST_EQUAL(RightmostDirection(90.), CarDirection::TurnRight, ());
   TEST_EQUAL(RightmostDirection(45.), CarDirection::TurnSlightRight, ());
   TEST_EQUAL(RightmostDirection(0.), CarDirection::GoStraight, ());
-  TEST_EQUAL(RightmostDirection(-20.), CarDirection::TurnSlightLeft, ());
-  TEST_EQUAL(RightmostDirection(-90.), CarDirection::TurnLeft, ());
-  TEST_EQUAL(RightmostDirection(-170.), CarDirection::TurnSharpLeft, ());
+  TEST_EQUAL(RightmostDirection(-20.), CarDirection::GoStraight, ());
+  TEST_EQUAL(RightmostDirection(-90.), CarDirection::GoStraight, ());
+  TEST_EQUAL(RightmostDirection(-170.), CarDirection::GoStraight, ());
```

Comment:
Could you explain why 90 degrees is TurnRight and -90 degrees is GoStraight?
The same with others .. Am I miss something with these angles?

Reply:
I was trying to explain it in RighmostDirection() definition:
```
      // For sure it's incorrect to give directions TurnLeft or TurnSlighLeft if we need the rightmost turn.
      // So GoStraight direction is given even for needed sharp left turn when other turns on the left and are more sharp.
      // The reason: the rightmost turn is the most straight one here.
```
If necessary I can make it more clear or move comments to \brief section.

Reply:
Ok, but could you also provide an example (in brief or description or when RighmostDirection/LeftmostDirection functions are called), who decided that the turn is "rightmost" or "leftmost"?
What does "rightmost/leftmost" actually mean?

Reply:
I see this block but can't understand what does it mean?
```
if (nodes.candidates.front().m_segment == firstOutgoingSeg)
        turn.m_turn = LeftmostDirection(turnAngle);
      else if (nodes.candidates.back().m_segment == firstOutgoingSeg)
        turn.m_turn = RightmostDirection(turnAngle);
      else
        turn.m_turn = intermediateDirection;
```

Reply:
Ah shit, if nodes.candidates are sorted by direction angle, then it makes sense :)

Reply:
Running road_access_test.cpp::RoadAccess_WayBlockedWhenStartButOpenWhenReach
FAILED
routing_tests/index_graph_tools.cpp:606 TEST(base::AlmostEqualAbs(pathWeight, expectedWeight, kEpsilon)) 10804 12802 [5: (0, 1) (1, 2) (2, 3) (3, 4) (4, 5) ]
Test took 1 ms

1  tests failed:
road_access_test.cpp::RoadAccess_WayBlockedWhenStartButOpenWhenReach

Reply:
What is your env (Linux/MacOS) and toolchain (gcc/clang --version)?

Reply:
BTW, try to run ./opening_hours_tests

Reply:
MacOS 12.13.1
Apple clang version 13.1.6 (clang-1316.0.21.2)
Target: x86_64-apple-darwin21.4.0
Thread model: posix

Boost just pulled.

*** 12 failures are detected in the test module "OpeningHours"

Reply:
Added:
```
// turnCandidates are sorted by angle from leftmost to rightmost.
if (turnCandidates.front().m_segment == firstOutgoingSeg)
```

Reply:
Strange, because I have exactly the same toolchain ..

Reply:
Do I need any additional data to run the tests?
I used "--filter=blob:limit=128k"

Reply:
No, you don't .. Could you show me:
cat 3party/boost/boost/version.hpp | grep "VERSION"
cat 3party/boost/boost/spirit/include/version.hpp | grep "VERSION"
cat 3party/boost/boost/phoenix/version.hpp | grep "VERSION"

Reply:
```
#define BOOST_VERSION 107900
#define BOOST_LIB_VERSION "1_79"

#define SPIRIT_VERSION 0x2059
#define SPIRIT_PIZZA_VERSION SUPER_HOT_SPANISH_SARDINES  // :-O

#define BOOST_PHOENIX_VERSION   0x3200    // 3.2.0
#define BOOST_PHOENIX_VERSION_NUMBER = BOOST_VERSION_NUMBER(3,2,0)
```